### PR TITLE
Change publisher

### PR DESCRIPTION
**Pull request by TommyDL-Infomaniak**

### CODE REVIEW

**Comment by TommyDL-Infomaniak** [comment]
The duplication of new code appears in the sample, it can be ignored.

File: README.md [hunk 1]
```diff
@@ -119,20 +127,28 @@ class MyAdapter(dataSet: List<String> = emptyList())
         viewHolder.itemText.text = item
     }
 
-    override fun getViewToTouchToStartDraggingItem(item: String, viewHolder: MyAdapter.ViewHolder, position: Int): View? {
+    override fun getViewToTouchToStartDraggingItem(
+        item: String,
+        viewHolder: MyAdapter.ViewHolder,
+        position: Int
```
Comment (KevinBoulongne) [inline README.md:133]:
```suggestion
        position: Int,
```

File: README.md [hunk 2]
```diff
@@ -144,48 +160,57 @@ mList.adapter = mAdapter
...
+Finally, create event listeners for the events you want to handle. For example, these are the
+listeners for actions of *swiping*, *dragging & dropping* and *scrolling*:
 
 ```kotlin
 private val onItemSwipeListener = object : OnItemSwipeListener<String> {
-    override fun onItemSwiped(position: Int, direction: OnItemSwipeListener.SwipeDirection, item: String): Boolean {
+    override fun onItemSwiped(
+        position: Int,
+        direction: OnItemSwipeListener.SwipeDirection,
+        item: String
```
Comment (KevinBoulongne) [inline README.md:212]:
```suggestion
        item: String,
```

File: README.md [hunk 3]
```diff
@@ -208,7 +233,10 @@ private val onListScrollListener = object : OnListScrollListener {
         // Handle change on list scroll state
     }
 
-    override fun onListScrolled(scrollDirection: OnListScrollListener.ScrollDirection, distance: Int) {
+    override fun onListScrolled(
+        scrollDirection: OnListScrollListener.ScrollDirection,
+        distance: Int
```
Comment (KevinBoulongne) [inline README.md:238]:
```suggestion
        distance: Int,
```

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/MainActivity.kt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package com.ernestoyaquello.dragdropswiperecyclerviewsample
+package com.github.infomaniak.dragdropswiperecyclerviewsample
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/MainActivity.kt:1]: Remove the `.github` in all packages.

File: drag-drop-swipe-recyclerview-sample/build.gradle [hunk 5]
```diff
@@ -5,7 +5,7 @@ android {
     compileSdk 34
 
     defaultConfig {
-        applicationId "com.ernestoyaquello.dragdropswiperecyclerviewsample"
+        applicationId "com.github.infomaniak.dragdropswiperecyclerviewsample"
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/build.gradle:8]: Remove the `.github` in the applicationId.

File: drag-drop-swipe-recyclerview-sample/build.gradle [hunk 6]
```diff
@@ -28,7 +28,7 @@ android {
         targetCompatibility rootProject.ext.javaVersion
     }
 
-    namespace 'com.ernestoyaquello.dragdropswiperecyclerviewsample'
+    namespace 'com.github.infomaniak.dragdropswiperecyclerviewsample'
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/build.gradle:31]: Remove the `.github` in the namespace.

File: drag-drop-swipe-recyclerview/build.gradle [hunk 7]
```diff
@@ -29,25 +23,32 @@ android {
...
+
     publishing {
         singleVariant("release") {
             withSourcesJar()
             withJavadocJar()
         }
     }
 
-    namespace 'com.ernestoyaquello.dragdropswiperecyclerview'
+    namespace 'com.github.infomaniak.dragdropswiperecyclerview'
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview/build.gradle:35]: Remove the `.github` in the namespace.

File: drag-drop-swipe-recyclerview/build.gradle [hunk 8]
```diff
@@ -29,25 +23,32 @@ android {
...
-ext {
-    PUBLISH_GROUP_ID = 'com.ernestoyaquello.dragdropswiperecyclerview'
-    PUBLISH_ARTIFACT_ID = 'drag-drop-swipe-recyclerview'
-    PUBLISH_VERSION = android.defaultConfig.versionName
+afterEvaluate {
+    publishing {
+        publications {
+            maven(MavenPublication) {
+                from components.findByName('release')
+                groupId = 'com.github.infomaniak.dragdropswiperecyclerview'
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview/build.gradle:48]:
```suggestion
                groupId = 'com.github.infomaniak'
```

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt [hunk 9]
```diff
@@ -104,7 +105,8 @@ class HorizontalListFragment : BaseListFragment() {
                 list.behindSwipedItemLayoutId = R.layout.behind_swiped_horizontal_list
 
                 // In XML: app:behind_swiped_item_custom_layout_secondary="@layout/behind_swiped_horizontal_list_secondary"
-                list.behindSwipedItemSecondaryLayoutId = R.layout.behind_swiped_horizontal_list_secondary
+                list.behindSwipedItemSecondaryLayoutId =
+                    R.layout.behind_swiped_horizontal_list_secondary
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt:109]: Revert ?

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt [hunk 10]
```diff
@@ -90,7 +90,8 @@ class HorizontalListFragment : BaseListFragment() {
                 list.behindSwipedItemIconSecondaryDrawableId = R.drawable.ic_archive_item
 
                 // In XML: app:behind_swiped_item_bg_color="@color/swipeBehindBackground"
-                list.behindSwipedItemBackgroundColor = ContextCompat.getColor(currentContext, R.color.swipeBehindBackground)
+                list.behindSwipedItemBackgroundColor =
+                    ContextCompat.getColor(currentContext, R.color.swipeBehindBackground)
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt:94]: Revert ?

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+        private var instance: IceCreamRepository? = null
+
+        fun getInstance(): IceCreamRepository {
+            if (instance == null)
+                instance = IceCreamRepository()
+
+            return instance as IceCreamRepository
+        }
+    }
+}
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:213]: Jump line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+
+    private fun ClosedRange<Int>.random() =
+        Random().nextInt((endInclusive + 1) - start) + start
+
+    companion object {
+        private var instance: IceCreamRepository? = null
+
+        fun getInstance(): IceCreamRepository {
+            if (instance == null)
+                instance = IceCreamRepository()
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:208]: One line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+
+    private fun generateIceCreamName() =
+        "${adjectives.shuffled().take(1)[0]} ${names.shuffled().take(1)[0]}"
+
+    private fun generateIceCreamPrice() = (80..500).random().toFloat() / 100f
+
+    private fun generateIceCreamBasicColor() = (80..210).random().toFloat() / 255f
+
+    private fun ClosedRange<Int>.random() =
+        Random().nextInt((endInclusive + 1) - start) + start
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:201]: One line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+            enhancedRed < enhancedGreen && enhancedRed < enhancedBlue -> enhancedRed -= 0.1f
+            enhancedGreen < enhancedRed && enhancedGreen < enhancedBlue -> enhancedGreen -= 0.1f
+            enhancedBlue < enhancedRed && enhancedBlue < enhancedGreen -> enhancedBlue -= 0.1f
+        }
+
+        return Triple(enhancedRed, enhancedGreen, enhancedBlue)
+    }
+
+    private fun generateIceCreamName() =
+        "${adjectives.shuffled().take(1)[0]} ${names.shuffled().take(1)[0]}"
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:194]: One line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 15]
```diff
@@ -0,0 +1,213 @@
...
+
+        addItem(iceCream)
+
+        return iceCream
+    }
+
+    private fun enhanceColorIntensity(
+        red: Float,
+        green: Float,
+        blue: Float
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:174]:
```suggestion
        blue: Float,
```

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 16]
```diff
@@ -0,0 +1,213 @@
...
+    override fun generateNewItem(): IceCream {
+        val iceCreamName = generateIceCreamName()
+        val iceCreamPrice = generateIceCreamPrice()
+        val red = generateIceCreamBasicColor()
+        val green = generateIceCreamBasicColor()
+        val blue = generateIceCreamBasicColor()
+        val (enhancedRed, enhancedGreen, enhancedBlue) = enhanceColorIntensity(red, green, blue)
+
+        val iceCream =
+            IceCream(iceCreamName, iceCreamPrice, enhancedRed, enhancedGreen, enhancedBlue)
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:164]: One line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 17]
```diff
@@ -0,0 +1,213 @@
...
+        "Snickers",
+        "Sports Mixture",
+        "Toffee Crisp",
+        "Topic",
+        "Trio",
+        "Twirl",
+        "Victory V",
+        "Wham Bar",
+        "Wine gum",
+        "Yorkie"
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:152]:
```suggestion
        "Yorkie",
```

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt [hunk 18]
```diff
@@ -0,0 +1,213 @@
...
+        "Savory",
+        "Sour",
+        "Spicy",
+        "Sticky",
+        "Strong",
+        "Sweet",
+        "Tangy",
+        "Tart",
+        "Tasteless",
+        "Tasty"
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt:35]:
```suggestion
        "Tasty",
```

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/model/IceCream.kt [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+class IceCream(
+    val name: String,
+    val price: Float,
+    val colorRed: Float,
+    val colorGreen: Float,
+    val colorBlue: Float
+) {
+
+    override fun toString() = name
+}
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/model/IceCream.kt:12]: Jump line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/config/local/AppConfig.kt [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+        isUsingStandardItemLayout = false,
+        isRestrictingDraggingDirections = false,
+        isDrawingBehindSwipedItems = true,
+        isUsingFadeOnSwipedItems = true
+    )
+)
+
+var currentListFragmentType = ListFragmentType.VERTICAL
+val currentListFragmentConfig
+    get() = listFragmentConfigurations[currentListFragmentType.index]
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/config/local/AppConfig.kt:45]: Jump line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/model/IceCream.kt [hunk 21]
```diff
@@ -0,0 +1,12 @@
+package com.github.infomaniak.dragdropswiperecyclerviewsample.data.model
+
+class IceCream(
+    val name: String,
+    val price: Float,
+    val colorRed: Float,
+    val colorGreen: Float,
+    val colorBlue: Float
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/model/IceCream.kt:8]:
```suggestion
    val colorBlue: Float,
```

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/MainActivity.kt [hunk 22]
```diff
@@ -120,14 +120,24 @@ class MainActivity : AppCompatActivity() {
         supportActionBar?.setDisplayHomeAsUpEnabled(false)
         supportActionBar?.setHomeButtonEnabled(false)
         binding.seeLogButton.visibility = View.VISIBLE
-        binding.fab.setImageDrawable(AppCompatResources.getDrawable(applicationContext, R.drawable.ic_new_item))
+        binding.fab.setImageDrawable(
+            AppCompatResources.getDrawable(
+                applicationContext,
+                R.drawable.ic_new_item
+            )
+        )
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/MainActivity.kt:128]: One line ?

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/MainActivity.kt [hunk 23]
```diff
@@ -120,14 +120,24 @@ class MainActivity : AppCompatActivity() {
...
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
         supportActionBar?.setHomeButtonEnabled(true)
         binding.seeLogButton.visibility = View.GONE
-        binding.fab.setImageDrawable(AppCompatResources.getDrawable(applicationContext, R.drawable.ic_clear_items))
+        binding.fab.setImageDrawable(
+            AppCompatResources.getDrawable(
+                applicationContext,
+                R.drawable.ic_clear_items
+            )
+        )
```
Comment (KevinBoulongne) [inline drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/MainActivity.kt:140]: One line ?

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt (hunk 9 above)
Comment (TommyDL-Infomaniak) [reply drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt]: It is too long to fit in one line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt (hunk 10 above)
Comment (TommyDL-Infomaniak) [reply drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/feature/managelists/view/HorizontalListFragment.kt]: It is too long to fit on one line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt (hunk 13 above)
Comment (TommyDL-Infomaniak) [reply drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt]: It is too long to fit on one line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt (hunk 14 above)
Comment (TommyDL-Infomaniak) [reply drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt]: It is too long to fit on one line.

File: drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt (hunk 16 above)
Comment (TommyDL-Infomaniak) [reply drag-drop-swipe-recyclerview-sample/src/main/java/com/github/infomaniak/dragdropswiperecyclerviewsample/data/source/IceCreamRepository.kt]: It is too long to fit on one line.

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**